packages/@aws-cdk/aws-stepfunctions/lib/states/state.ts

```diff
@@ -82,13 +82,26 @@ export abstract class State extends CoreConstruct implements IChainable {
    * Add a prefix to the stateId of all States found in a construct tree
    */
   public static prefixStates(root: IConstruct, prefix: string) {
+    const visited = new Set();
     const queue = [root];
     while (queue.length > 0) {
       const el = queue.splice(0, 1)[0]!;
       if (isPrefixable(el)) {
         el.addPrefix(prefix);
       }
-      queue.push(...Node.of(el).children);
+      if (isBranchable(el)) {
+        addToQueue(el.branchableBranches);
+      }
```

Comment:
Remind me why this is necessary?

It seems like otherwise it wouldn't prefix states which aren't the construct's children, but that is fine...?

Reply:
not necessary unfortunately :(. deleted.